src/node_http_parser.cc

```diff
@@ -616,22 +697,29 @@ class Parser : public AsyncWrap, public StreamListener {
     if (got_exception_)
       return scope.Escape(Local<Value>());
 
-    Local<Integer> nparsed_obj = Integer::New(env()->isolate(), nparsed);
+    Local<Integer> nread_obj = Integer::New(env()->isolate(), nread);
+
     // If there was a parse error in one of the callbacks
     // TODO(bnoordhuis) What if there is an error on EOF?
-    if (!parser_.upgrade && nparsed != len) {
```

Comment:
Hey @indutny, you got rid of the `nparsed != len` check here entirely for http_parser. Is this safe? Are we guaranteed to get an `err != HPE_OK` if this fails within http_parser?

Reply:
It is alright, because it is either an error or `parser_.upgrade` when the `nparsed != len` in http_parser.